### PR TITLE
CloudTrail: tighten IAM permissions, enable encryption

### PR DESCRIPTION
**Pull request by jakemulley**
This tightens IAM permissions for access to the AWS KMS key and enables encryption for CloudTrail and SNS.

It also resolves some AWS Config non-compliancy issues.

### CODE REVIEW

File: modules/cloudtrail/main.tf [hunk 1]
```diff
@@ -78,7 +83,7 @@ resource "aws_cloudwatch_log_group" "cloudtrail" {
   tags       = var.tags
 }
 
-resource "aws_cloudwatch_log_stream" "cloudtrail_stream" {
+resource "aws_cloudwatch_log_stream" "cloudtrail-stream" {
```
Comment (davidkelliott) [inline modules/cloudtrail/main.tf:86]: I had to look at this 10 times before I spotted the difference 😂

**Review by davidkelliott — APPROVED** [review]
👍

File: modules/cloudtrail/main.tf (hunk 1 above)
Comment (jakemulley) [reply modules/cloudtrail/main.tf:92]: 😆 I realised I used dashes everywhere else, it took me a while to realise why Terraform was telling me it'll destroy and recreate the same resource!

**Review by davidkelliott — APPROVED** [review]
👍

**Pull request merged**